### PR TITLE
[DOWNSTREAM] Cyrillic Support

### PR DESCRIPTION
**Pull request by larentoun**
## What Does This PR Do

Adds ciryllic support from https://github.com/ss220club/Paradise/commit/cbdad6f11f0e0e16694cdb60f8ccb3d6dd23fe92

Changes all `sanitize(copytext(` with `sanitize(copytext_char(` and some other things I'm too tired to write. Mostly just adds `_char` prefix.

## Why It's Good For The Game

Russian downstream need this to minimize direct changes to the code for their direct fork. 

## Testing

image

## Changelog
:cl:
add: Cyrillic Support for a lot of things, mostly names and capitalize
/:cl:

### CODE REVIEW

**Comment by Furrior** [comment]
Here we go

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -256,14 +256,14 @@
 //Returns a string with reserved characters and spaces before the first letter removed
 /proc/trim_left(text)
 	for(var/i = 1 to length(text))
-		if(text2ascii(text, i) > 32)
+		if(text2ascii(text, i) != 32)
```
Comment (lewcc) [inline code/__HELPERS/text.dm:259]: This line (and the similar one in trim_right) is what I'm not quite sure of. Are such control characters used in cyrillic?

**Review by lewcc — COMMENTED** [review]
Mostly looks good, I can't imagine when we would want to use base copytext anyway in this case. I'm curious if it has any sort of performance impact, though I'd imagine it'd probably be negligible.

**Comment by larentoun** [comment]
> Mostly looks good, I can't imagine when we would want to use base copytext anyway in this case. I'm curious if it has any sort of performance impact, though I'd imagine it'd probably be negligible.

DM ref says it does have less performance, but I don't know how much

File: code/__HELPERS/text.dm (hunk 1 above)
Comment (larentoun) [reply code/__HELPERS/text.dm:259]: It was changed to that in the oldpara220, so I'm not quite sure. Was mostly porting the cyrillic fix from the old host-coder

Comment (larentoun) [reply code/__HELPERS/text.dm]: But it doesn't seem to care about it, removed it

**Comment by farie82** [comment]
Sorry for getting back this late to the PR.

We don't see this feature being of use on our codebase due to our main server only permitting the English language for players. Downstreams are free to implement this feature on their own if they so choose.
Due to this, the PR will be closed.

**Pull request closed without merging**